libs/cdk/render-strategies/docs/basic-strategies.md

```diff
@@ -0,0 +1,119 @@
+## Basic Strategies
+
+### Concepts
+
+#### Render work
+
+To apply changes to a component template we need to re-evaluate the template. Internally this is done by calling `???`.
+This method will execute whenever a component's template is re-evaluated through `async`, `push`, `ChangedDetectorRef.detectChanges` or a structural directive template is re-evaluated through `EmbeddedView.detectChanges`
+
+This process can be pretty time consuming and directly depends on the following factors:
+- HTML size (only for init and destroy and bundle-size)
+- JS size (only for init and destroy and bundle-size)
+- number of event bindings (only for init and destroy)
+- number of template bindings/expressions
+- number and size of child-components
+- number of directives (especially ngFor and nested structures)
+
+Some of the problems related to work of Angular are:
+
+**Out of bound change detection:**
+If we perform this re-evaluation without any visual change for the user (over-rendering) we introduce noticable performance degradations.
+
+The out of bound change detection can be caused through: 
+  - Zone pollution
+  - Missing ChangeDetectionStrategy.OnPush 
+  - Component template projection
+  - Pull-based rendering processes 
+  
+**Out of bound template evaluation:**
+If we perform a re-evaluation of a single property in the template any other expression/binding also gets reevaluated. 
+Again over-rendering introduces noticable performance degradations.
+
+The out of bound template evaluation can be caused through:
+  - Pull-based rendering processes
+  - any reactive change throught `async`
+  - any call to `cdRef.detectChanges`
+
+
+**Work performed for out of viewport content:**
+If we perform a re-evaluation or even re-rendering of the DOM of elements outside of the viewport we perform useless work for the user.
+This will pollute the main thread and reduced time for more important content to get rendered.
+
+The re-evaluation or browser re-rendreing can be caused by:
+- Bad style changes
+- Big LCP elements
+- Large amount of content
+
+#### Local vs global
+
+#### Pull vs push based
```

Comment:
Two titles instead of one here. Missing content?

Reply:
yes